### ISSUE TITLE
`$SHELL` not updated when using `kitty +kitten ssh`

### ISSUE THREAD

**Issue by nalzok** [kovidgoyal/kitty#7765]
**Describe the bug**
In a session initiated by `kitty +kitten ssh`, `$SHELL` still points to `/bin/sh` after `chsh -s $(which fish)`. This confuses programs like `tmux`.

**To Reproduce**

1. On the OpenBSD host, set `fish` as the login shell with `chsh -s $(which fish)`.
2. Log in using `kitty +kitten ssh magicbane`.
```fish
$ kitty +kitten ssh magicbane
Welcome to fish, the friendly interactive shell
Type help for instructions on how to use fish
qys@magicbane ~> echo $SHELL
/bin/sh
```
3. For comparison, log in using `/usr/bin/ssh`. Not only didn't `kitty +kitten ssh magicbane` output MOTD, but it also does not show the crucial `You have new mail.` line.
```fish
$ /usr/bin/ssh magicbane
Last login: Wed Aug 21 10:32:29 on ttyp0 from 100.94.151.103
OpenBSD 7.6-beta (GENERIC.MP) #270: Tue Aug 20 10:51:30 MDT 2024

Welcome to OpenBSD: The proactively secure Unix-like operating system.

Please use the sendbug(1) utility to report bugs in the system.
Before reporting a bug, please try to reproduce it with the latest
version of the code.  With bug reports, please try to ensure that
enough information to reproduce the problem is enclosed, and if a
known fix for it exists, include that as well.

You have new mail.
Welcome to fish, the friendly interactive shell
Type help for instructions on how to use fish
qys@magicbane ~> echo $SHELL
/usr/local/bin/fish
```

**Screenshots**
<img width="1388" alt="image" src="https://github.com/user-attachments/assets/8c4024c3-80f2-4e90-bf58-9d28976ae3f1">

**Environment details**
```
kitty 0.35.2 created by Kovid Goyal
Darwin Qingyaos-MBP.lan 22.6.0 Darwin Kernel Version 22.6.0: Mon Jun 24 01:25:37 PDT 2024; root:xnu-8796.141.3.706.2~1/RELEASE_X86_64 x86_64
ProductName:		macOS ProductVersion:		13.6.9 BuildVersion:		22G830
OpenGL: '4.1 INTEL-20.7.2' Detected version: 4.1
Frozen: False
Paths:
  kitty: /Applications/MacPorts/kitty.app/Contents/MacOS/kitty
  base dir: /Applications/MacPorts/kitty.app/Contents/Resources/kitty
  extensions dir: /Applications/MacPorts/kitty.app/Contents/Resources/kitty/kitty
  system shell: /opt/local/bin/fish
Loaded config files:
  /Users/qys/.config/kitty/kitty.conf

Config options different from defaults:
cursor_blink_interval              0.0
enable_audio_bell                  False
font_family                        PragmataPro Mono
font_features:
{'PragmataProMono-Regular': ('+calt', '+cv03')}
font_size                          16.0
macos_option_as_alt                3
macos_quit_when_last_window_closed True
shell_integration                  frozenset({'no-cursor'})
tab_bar_edge                       1
tab_bar_style                      separator
tab_title_template                 {fmt.fg.red}{bell_symbol}{activity_symbol}{fmt.fg.tab}{sup.index}{title}
visual_bell_duration               0.1
Changed mouse actions:
	right press ungrabbed →  mouse_select_command_output
Removed shortcuts:
	cmd+, →  edit_config_file
	cmd+enter →  new_window
	cmd+h →  hide_macos_app
	cmd+k →  clear_terminal to_cursor active
	cmd+m →  minimize_macos_window
	cmd+n →  new_os_window
	cmd+q →  quit
	cmd+t →  new_tab
	cmd+w →  close_tab
	ctrl+cmd+, →  load_config_file
	ctrl+cmd+f →  toggle_fullscreen
	opt+cmd+h →  hide_macos_other_apps
	opt+cmd+r →  clear_terminal reset active
	opt+cmd+s →  toggle_macos_secure_keyboard_entry
	shift+cmd+/ →  open_url https://sw.kovidgoyal.net/kitty/
	shift+cmd+[ →  previous_tab
	shift+cmd+] →  next_tab
	shift+cmd+d →  close_window
	shift+cmd+w →  close_os_window
Changed shortcuts:
	cmd+0 →  goto_tab    10
	cmd+1 →  goto_tab    1
	cmd+2 →  goto_tab    2
	cmd+3 →  goto_tab    3
	cmd+4 →  goto_tab    4
	cmd+5 →  goto_tab    5
	cmd+6 →  goto_tab    6
	cmd+7 →  goto_tab    7
	cmd+8 →  goto_tab    8
	cmd+9 →  goto_tab    9

Important environment variables seen by the kitty process:
	PATH                                /Applications/MacPorts/kitty.app/Contents/MacOS:/usr/bin:/bin:/usr/sbin:/sbin
	LANG                                en_US.UTF-8
	SHELL                               /opt/local/bin/fish
	DISPLAY                             /private/tmp/com.apple.launchd.TzcUdzz3Gc/org.xquartz:0
	USER                                qys
	LC_CTYPE                            UTF-8
```

**Additional context**
I can reproduce the problem with `kitty --config NONE`.

In case it matters, the client is running Kitty 0.35.2 on macOS 13.6.9, while the remote host (let's call it `magicbane`) is running OpenBSD-current with no Kitty installed, but `~/.local/share/kitty-ssh-kitten/kitty/bin/kitten --version` shows `kitten 0.36.0 created by Kovid Goyal`

**Comment by kovidgoyal** [comment]
chsh doesnt change the value of $SHELL. This has nothing to do with the
ssh kitten.

**Comment by nalzok** [comment]
Hmmm, these two commands report different `$SHELL` despite SSHing to the same target. What else can explain the discrepancy between `kitty +kitten ssh` and `/usr/bin/ssh`? Surely `kitty +kitten ssh` is doing some extra step that `/usr/bin/ssh` is not doing.

**Comment by kovidgoyal** [comment]
the ssh kitten will execute whatever the default login sheel for the
user is, unless you have configured it not too. If it is not doing that,
then you have configured it to use a different shell.

**Comment by gkshenaut** [comment]
I recently encountered a related kitty issue. On MacOS, I was migrating from ksh and .kshrc to zsh and .zshrc. The change worked fine with mac Terminal, but kitty kept sourcing .kshrc. I eventually tracked it down to a somehow kitty-cached ENV=<home>/.kshrc, from before the chsh. After futzing for too long, I finally rebooted, which evidently reset the cache, so kitty started sourcing .zshrc and the world became a better place.
